pkg/api/object.go

```diff
@@ -18,19 +18,49 @@ func getObjectIdentifier(obj types.Object, singleObjectOperation bool) (string,
 	}
 
 	objectStructType := objectType.Elem()
-	numFields := objectStructType.NumField()
+	return findIdentifierInStruct(objectStructType, reflect.ValueOf(obj).Elem(), singleObjectOperation)
+}
+
+func findIdentifierInStruct(t reflect.Type, v reflect.Value, singleObjectOp bool) (string, error) {
+	// we also use this to track if we found an identifier already
+	var returnIdentifier *string
 
+	numFields := t.NumField()
+
+fields:
 	for i := 0; i < numFields; i++ {
-		field := objectStructType.Field(i)
+		field := t.Field(i)
+
+		if field.Anonymous {
+			embeddedType := field.Type
+			embeddedValue := v.Field(i)
+
+			for embeddedType.Kind() == reflect.Ptr {
+				embeddedType = embeddedType.Elem()
+				embeddedValue = embeddedValue.Elem()
+			}
+
+			if embeddedType.Kind() == reflect.Struct {
+				if ret, err := findIdentifierInStruct(embeddedType, embeddedValue, singleObjectOp); err == nil {
```

Comment:
Just out of curiosity: Is this DoS-able by having a (malicious) self-referencing or circular struct definition?

Reply:
Plain structs embedded in itself does not compile: https://play.golang.org/p/AM5EbV_OG3m

Struct embedding a pointer to itself compiles and would indeed be an endless loop here, but this can only be done by authors or users of this package - not by data returned from the engine or anything else at runtime, so I think this is only a theoretical problem not worth the complexity of checking that.